picotest_helpers/src/lib.rs

```diff
@@ -119,8 +121,8 @@ impl Cluster {
             .build()?;
 
         debug!("Starting the cluster with parameters {params:?}");
-        let intances: Vec<PicodataInstance> = pike::cluster::run(&params)?;
-        self.instances.replace(intances);
+        let mut intances: Vec<PicodataInstance> = pike::cluster::run(&params)?;
+        std::mem::swap(&mut self.instances, &mut intances);
```

Comment:
Предлагаю здесь добавить debug_assert!, чтобы точно отловить нарушение инварианта хотябы в дебаге
```suggestion
        debug_assert!(self.instances.is_empty(), "trying to replace already running clister?");
        std::mem::swap(&mut self.instances, &mut intances);
```